services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/CrtErrorHandler.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.transfer.s3.internal;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.s3.CrtS3RuntimeException;
+import software.amazon.awssdk.utils.StringUtils;
+
+import java.util.Optional;
+
+@SdkInternalApi
+public class CrtErrorHandler {
+
+    /**
+     * This class transform a CRTRunTimeException to Meaningful SdkServiceException.
+     * CrtS3RuntimeException are the exceptions generated due to failures in CRTClient due to S3 Service errors.
+     *
+     * @param crtRuntimeException Exception that is thrown by CRTClient.
+     * @return
+     */
+    public Exception transformException(Exception crtRuntimeException) {
+        Optional<CrtS3RuntimeException> crtS3RuntimeExceptionOptional = getCrtS3RuntimeException(crtRuntimeException);
+
+        return crtS3RuntimeExceptionOptional.isPresent()
```

Comment:
Minor: instead of checking if it's present and doing `crtS3RuntimeExceptionOptional.get()`, we can do the following:
```
return crtS3RuntimeExceptionOptional.filter(xxxx).orElse(crtRuntimeException)
```

Can we break up the code block within `filter(xx)`? It's a bit hard to read

Reply:
Done